### PR TITLE
feat: Add endpoint to transfer cash between accounts

### LINKED ISSUE #2446: Add endpoint to transfer cash between accounts
This is a follow up task of #2432.

### Todo

- [ ] Create endpoint `POST api/v1/account/transfer-balance` using `TransferBalanceDto`
- [ ] Add logic which calls [updateAccountBalance()](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/account/account.service.ts#L217) for both affected accounts 
- [ ] Enable transfer button [here](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/components/accounts-table/accounts-table.component.html#L1)
- [ ] Call endpoint on submit from client

### PR DESCRIPTION
**Pull request by Dhoni77**
Closes #2446

- Enabled transfer button on UI
- Created transfer api on client
- Implemented post transfer balance on submit
- Created the server side end point

### CODE REVIEW

**Comment by Dhoni77** [comment]
cc @dtslvr 
Hi, I just had a doubt regarding the updateAccountBalance() logic for updating the account balance.
Since we are not reducing the amount from the fromAccount, instead we are adding the entered amount to both the from and to accounts.
Is this expected or should I make any logic changes ?

File: apps/client/src/app/services/data.service.ts [hunk 1]
```diff
@@ -471,6 +472,18 @@ export class DataService {
     return this.http.post<OrderModel>(`/api/v1/account`, aAccount);
   }
 
+  public postTransferAccountBalance({
```
Comment (dtslvr) [inline apps/client/src/app/services/data.service.ts:475]: Can you move down ⬇️ to keep the alphabetical order?

File: apps/client/src/app/pages/accounts/accounts-page.component.ts [hunk 2]
```diff
@@ -298,9 +298,18 @@ export class AccountsPageComponent implements OnDestroy, OnInit {
           console.log(
```
Comment (dtslvr) [inline apps/client/src/app/pages/accounts/accounts-page.component.ts:298]: Please remove this

File: apps/api/src/app/account/account.controller.ts [hunk 3]
```diff
@@ -154,6 +156,40 @@ export class AccountController {
...
+        StatusCodes.FORBIDDEN
+      );
+    }
+
+    const today = resetHours(new Date());
+    const userCurrency = this.request.user.Settings.settings.baseCurrency;
+
+    await this.accountService.updateAccountBalance({
+      accountId: accountIdFrom,
+      amount: balance,
```
Comment (dtslvr) [inline apps/api/src/app/account/account.controller.ts:178]: > Is this expected or should I make any logic changes?

Adding the negative balance here should work:

```
amount: -balance,
```

File: apps/api/src/app/account/account.controller.ts [hunk 4]
```diff
@@ -154,6 +156,40 @@ export class AccountController {
...
+  ) {
+    if (
+      !hasPermission(this.request.user.permissions, permissions.updateAccount)
+    ) {
+      throw new HttpException(
+        getReasonPhrase(StatusCodes.FORBIDDEN),
+        StatusCodes.FORBIDDEN
+      );
+    }
+
```
Comment (dtslvr) [inline apps/api/src/app/account/account.controller.ts:172]: For security reasons, a check is needed that the accounts with id `accountIdFrom` and `accountIdTo` belong to the user.

Can you get the accounts like this:
```
this.accountService.getAccounts(this.request.user.id);
```
And throw a new `NOT_FOUND` exception if the check fails?

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks for your contribution, @Dhoni77 🙂 I have added some comments.

File: apps/client/src/app/pages/accounts/accounts-page.component.ts (hunk 2 above)
Comment (Dhoni77) [reply apps/client/src/app/pages/accounts/accounts-page.component.ts:298]: done

File: apps/api/src/app/account/account.controller.ts (hunk 4 above)
Comment (Dhoni77) [reply apps/api/src/app/account/account.controller.ts:172]: yes

**Review by dtslvr — APPROVED** [review]
Thank you for the modifications. I have slightly refactored the code, ready to merge now.

**Pull request merged**